src/mito2/src/region/opener.rs

```diff
@@ -120,6 +126,7 @@ impl RegionOpener {
         match self.maybe_open(config, wal).await {
             Ok(Some(region)) => {
                 let recovered = region.metadata();
+                // Safety: must exist.
                 // Checks the schema of the region.
                 let expect = self.metadata.as_ref().unwrap();
```

Comment:
Maybe we can add pass the `metadata` to `create_or_open` so we don't need to keep it as a field in the opener.